src/app/features/notification/notification.service.ts

```diff
@@ -45,20 +48,74 @@ export class NotificationService {
    * Requests permission for Firebase messaging and retrieves the token.
    * Logs the token if successful, otherwise logs an error or info message.
    */
-  async getFcmToken(): Promise<any> {
+  async getFcmToken(): Promise<string> {
     try {
+      const existingNotificationToken = this.getNotificationTokenFromCookie();
+
       const notificationToken = await getToken(this.messaging, {
         vapidKey: environment.firebaseConfig.vapidKey,
       });
-      if (notificationToken) {
-        // TODO: Need to Implement the logic to save the FCM token in the Cookie and update in the backend API.
-        return notificationToken;
+
+      if (notificationToken === existingNotificationToken) {
+        return existingNotificationToken;
       }
+
+      this.setCookie(
+        this.NOTIFICATION_TOKEN_COOKIE_NAME,
+        notificationToken,
+        30,
+      );
+
+      this.registerNotificationToken(notificationToken);
+
+      return notificationToken;
     } catch (err) {
       Logging.error("An error occurred while retrieving token: ", err);
     }
   }
 
+  /**
+   * Registers the device with the backend using the FCM token.
+   * @param notificationToken - The FCM token for the device.
+   * @param deviceName - The name of the device.
+   */
+
+  private registerNotificationToken(
+    notificationToken: string,
+    deviceName: string = "web",
+  ) {
+    const payload = { fcmToken: notificationToken, deviceName };
+
+    try {
+      this.httpClient
+        .post(this.DEVICE_NOTIFICATION_API_URL, payload)
+        .subscribe();
+    } catch (err) {
+      Logging.error("Failed to register device: ", err);
+    }
+  }
+
+  /**
+   * Retrieves the value of a cookie by name.
+   * @param name - The name of the cookie.
+   */
+  getNotificationTokenFromCookie() {
+    const cookies = new URLSearchParams(document.cookie.replace(/; /g, "&"));
```

Comment:
While implementing this, I explored the best way to store and retrieve data in cookies using Angular. I reviewed several articles, and most suggested using third-party packages. However, I noticed that some built-in methods for working with cookies. So, instead of installing a package, I decided to implement the functionality using default methods.

Reply:
Let's either create a separate CookieService for this or indeed use a third-party package. I would also like to have a method to delete a cookie rather than just setting it to "null" above in the error case.